pkg/controller/service/controller.go

```diff
@@ -103,12 +98,13 @@ type serviceCache struct {
 // Controller keeps cloud provider service resources
 // (like load balancers) in sync with the registry.
 type Controller struct {
-	cloud               cloudprovider.Interface
-	knownHosts          []*v1.Node
-	servicesToUpdate    []*v1.Service
-	kubeClient          clientset.Interface
-	clusterName         string
-	balancer            cloudprovider.LoadBalancer
+	cloud            cloudprovider.Interface
+	knownHosts       []*v1.Node
+	servicesToUpdate []*v1.Service
+	kubeClient       clientset.Interface
+	clusterName      string
+	balancer         cloudprovider.LoadBalancer
+	// TODO(#85155): Stop relying on this and remove the cache completely.
```

Comment:
BTW created https://github.com/kubernetes/kubernetes/issues/85155 for cleaning up the cache layer eventually.